components/raftstore-v2/src/operation/ready/snapshot.rs

```diff
@@ -259,6 +259,10 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
                 meta.region_read_progress
                     .insert(region_id, self.read_progress().clone());
             }
+
+            let region_state = self.raft_group().store().region_state().clone();
+            self.storage_mut().set_region_state(region_state);
```

Comment:
The state should be set in `apply_snapshot`.

Reply:
Fine, I will propose another PR for it.